t/api/04-jobs.t

```diff
@@ -1482,7 +1482,57 @@ subtest 'marking job as done' => sub {
         $t->json_is('/job/result' => INCOMPLETE, 'result set');
         $t->json_is('/job/reason' => $cache_failure_reason, 'reason set');
         $t->json_is('/job/state' => DONE, 'state set');
-        $t->json_like('/job/clone_id' => qr/\d+/, 'job cloned when reason does matches configured regex');
+        $t->json_like('/job/clone_id' => qr/\d+/, 'job cloned when reason does match configured regex');
+
+        subtest 'auto_clone limits' => sub {
+            $schema->txn_begin;
+            local $t->app->config->{global}{auto_clone_limit} = 5;
+            my @jobs;
+            my $backend_reason = 'backend died: VNC Connection refused';
+            for my $i (reverse 10 .. 20) {
+                my $clone_id = $i < 20 ? $i + 1 : undef;
```

Comment:
I suggest for the sake of efficient testing to limit this to "auto_clone_limit+1" and the auto_clone_limit could be the most minimal. I guess that would be 3+1 considering you want to build a chain of "same, unrelated, same"